string floor
int32 target_number
---
bool flag